pytorch_lightning/trainer/supporters.py

```diff
@@ -68,16 +68,16 @@ def reset(self, window_length: Optional[int] = None) -> None:
     def last(self):
         """Get the last added element."""
         if self.last_idx is not None:
-            return self.memory[self.last_idx]
+            return self.memory[self.last_idx].float()
 
     def append(self, x):
         """Add an element to the accumulator."""
         if self.memory is None:
-            self.memory = torch.zeros(self.window_length, *x.shape)
-
-        # ensure same device and type
-        if self.memory.device != x.device or self.memory.type() != x.type():
-            x = x.to(self.memory)
+            # tradeoff memory for speed by keeping the memory on device
+            self.memory = torch.zeros(self.window_length, *x.shape, device=x.device)
+        else:
+            # ensure same device and type
+            self.memory = self.memory.to(x)
```

Comment:
is there any reason why dtype is not specified here but in the next iteration under `else` condition?
```suggestion
            self.memory = torch.zeros(self.window_length, *x.shape, device=x.device, dtype=x.dtype)
```